datafusion/physical-expr/src/planner.rs

```diff
@@ -294,43 +292,18 @@ pub fn create_physical_expr(
                             input_schema,
                             execution_props,
                         ),
-                        // TODO refactor the logic of coercion the data type
```

Comment:
cc @liukun4515

Reply:
This pr resolve some TODO inList for me.
it make sense to me